### ISSUE TITLE
Run geth failed

### ISSUE THREAD

**Issue by rjl493456442** [ethereum/go-ethereum#3126]
#### System information

Geth version: `geth version`  : 1.4.17
OS & Version: Windows/Linux/OSX  : Linux
Commit hash : (if `develop`)  : 5a6008e

#### Actual behaviour

`Out Of Memory` Occurred frequently when i run the `geth`.  Here is some error log

```
 I1012 02:56:05.031681 core/blockchain.go:975] imported 2048 block(s) (0 queued 0 ignored) including 15673 txs in 56.217819928s. #1717531 [6ce8e172 / 87efe14f]
fatal error: runtime: out of memory

runtime stack:
runtime.throw(0xd15ce3, 0x16)
    /usr/lib/go-1.7/src/runtime/panic.go:566 +0x95
runtime.sysMap(0xc468c70000, 0x2000000, 0x0, 0x1540d38)
    /usr/lib/go-1.7/src/runtime/mem_linux.go:219 +0x1d0
runtime.(*mheap).sysAlloc(0x1527680, 0x2000000, 0x1)
    /usr/lib/go-1.7/src/runtime/malloc.go:407 +0x37a
runtime.(*mheap).grow(0x1527680, 0x1000, 0x0)
    /usr/lib/go-1.7/src/runtime/mheap.go:726 +0x62
runtime.(*mheap).allocSpanLocked(0x1527680, 0x1000, 0x7f4fd67ea068)
    /usr/lib/go-1.7/src/runtime/mheap.go:630 +0x4f2
runtime.(*mheap).alloc_m(0x1527680, 0x1000, 0x7f0100000000, 0x435c4e)
    /usr/lib/go-1.7/src/runtime/mheap.go:515 +0xe0
runtime.(*mheap).alloc.func1()
    /usr/lib/go-1.7/src/runtime/mheap.go:579 +0x4b
runtime.systemstack(0x7f4fecac5dd8)
    /usr/lib/go-1.7/src/runtime/asm_amd64.s:314 +0xab
runtime.(*mheap).alloc(0x1527680, 0x1000, 0x10100000000, 0x7f4fecac5e58)
    /usr/lib/go-1.7/src/runtime/mheap.go:580 +0x73
runtime.largeAlloc(0x2000000, 0x7f4fecac5e01, 0x46893a)
    /usr/lib/go-1.7/src/runtime/malloc.go:774 +0x93
runtime.mallocgc.func1()
    /usr/lib/go-1.7/src/runtime/malloc.go:669 +0x3e
runtime.systemstack(0xc42001ea00)
    /usr/lib/go-1.7/src/runtime/asm_amd64.s:298 +0x79
runtime.mstart()
    /usr/lib/go-1.7/src/runtime/proc.go:1079
```

Should I configure the linux machine memory limit ? 

Here is my system configure

```
core file size          (blocks, -c) 0
data seg size           (kbytes, -d) unlimited
scheduling priority             (-e) 0
file size               (blocks, -f) unlimited
pending signals                 (-i) 32043
max locked memory       (kbytes, -l) 64
max memory size         (kbytes, -m) unlimited
open files                      (-n) 500000
pipe size            (512 bytes, -p) 8
POSIX message queues     (bytes, -q) 819200
real-time priority              (-r) 0
stack size              (kbytes, -s) 8192
cpu time               (seconds, -t) unlimited
max user processes              (-u) 32043
virtual memory          (kbytes, -v) unlimited
file locks                      (-x) unlimited
```

There is no limitation in memory

**Comment by lihengl** [comment]
Hi there, how much memory do you have on the machine? I'm maintaining four `1.4.17` clients and these days they all require machines with more than 8G of memory to run without crashing every now and then. Just saying you're not alone.

**Comment by rjl493456442** [comment]
My machine has 8GB physical memory exactly.

**Comment by lihengl** [comment]
Occasionally it peaks through 8G. That's why it crashes on your machine.

**Comment by JettyZ** [comment]
8GB requirement?  Will this be fixed?

**Comment by fjl** [comment]
The huge memory usage is the result of yet another DoS attack targeting geth. We are working on a fix.

**Comment by Mordong** [comment]
useless...

**Comment by karalabe** [comment]
@Mordong Thank you very much for your support :) It's really comforting to know that people understand and feel compassion during hard times ;)

**Comment by fjl** [comment]
This should be resolved in geth 1.4.18.
